crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -274,7 +280,8 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
         // ------------------- BUSINESS LOGIC -------------------
         let table_id = require_table_id(&table, table_id)?;
         let mut t = C::Transaction::begin_read(state.v1_state.catalog).await?;
-        let mut metadatas = C::load_tables(warehouse_id, vec![table_id], t.transaction()).await?;
+        let mut metadatas =
+            C::load_tables(warehouse_id, vec![table_id], false, t.transaction()).await?;
```

Comment:
I think it would be clearer to add a variable let include_deleted instead of the false here. We can re-use it also for the ListFlags above.